lib/colorado_lottery.rb

```diff
@@ -0,0 +1,57 @@
+class ColoradoLottery
+  attr_reader :registered_contestants, :current_contestants, :winners
+
+  def initialize()
+    @registered_contestants = {}
+    @current_contestants = {}
+    @winners = []
+  end
+
+  def interested_and_18?(contestant, game)
+    if contestant.age >= 18
+      contestant_interest = false
+      if contestant.game_interests.each do |interest|
+        contestant_interest = true if interest == game.name
+      end
+      contestant_interest
+    end
+  else
+    false
```

Comment:
good job getting this to work-- however, this involves a lot of nested ifs and some unnecessary iteration, which makes it both inefficient in terms of performance as well as difficult to read.  something like this would be ideal:

```
  def interested_and_18?(contestant, game)
      contestant.age >= 18 && contestant.game_interests.include?(game.name)
  end
```